gtsam/discrete/DiscreteConditional.cpp

```diff
@@ -203,7 +248,7 @@ size_t DiscreteConditional::sample(const DiscreteValues& parentsValues) const {
   static mt19937 rng(2);  // random number generator
 
   // Get the correct conditional density
-  ADT pFS = choose(parentsValues);  // P(F|S=parentsValues)
+  ADT pFS = Choose(*this, parentsValues);  // P(F|S=parentsValues)
 
   // TODO(Duy): only works for one key now, seems horribly slow this way
   assert(nrFrontals() == 1);
```

Comment:
Is this still true?

Reply:
This is true. I replaced this with an exception, though.